### ISSUE TITLE
Questions about the attribute recognition

### ISSUE THREAD

**Issue by ccJia** [xh-liu/HydraPlus-Net#3]
Hi Liu,  
    I tried to duplicate your paper, but I met some questions about the attribute recognition. My net is composed of "res-net" and a "SigmoidCrossEntropyLoss" layer. And the  attributes "trousers" and "Age 18-60" occupy a big proportion  in "PA-100K" dataset.  And it would lead a phenomenon  that  when I enter a part of  a picture the net could also give the labels include "trousers" and "Age 18-60"(I just want a single attribute ).  Could you give me some advice to eliminate these distractions?  Thank you.

**Comment by xh-liu** [comment]
Thank you for your question! It is a common phenomenon that the network tends to be affected by majority labels in classification. To avoid the effect of it we used a weighted sigmoid cross entropy loss, following previous work. For example, if the positive ratio is too large for attribute "trousers", we lower that loss penalty for positive samples and use higher loss weight for negative attributes. The formulation of the loss function can be found in the paper.

**Comment by ccJia** [comment]
Thank you for your reply.  I'll try this method .^-^

**Comment by JohnHush** [comment]
@ccJia hey, dude, are you re implementing the paper in Caffe? lol:)

**Comment by ccJia** [comment]
@JohnHush  Yes , I am implementing the paper. And I just finished the attribute extracting . I want add the attention net in it .

**Comment by JohnHush** [comment]
@ccJia me, too. I'm wondering how to add attention in the M-NET, need to build new layer?? do you ha ve some suggestion?

**Comment by ccJia** [comment]
@JohnHush We stand on the same starting line. I just asked my friend about this. ^-^ If there is any good idea, I will contact you .

**Comment by JohnHush** [comment]
@ccJia that's cool, recently I've implemented DeepMAR, I supposed it's basically the same as M-NET without Attention, keep in touch

**Comment by Li1991** [comment]
Hi, have you re-implemented this paper? I am doubt about how to make use of attention. Can you give some suggestions? Thank you. @ccJia

**Comment by JohnHush** [comment]
@kclch ,hi, yes i have reproduced DeepMar, just follow the instruction in the paper"Multi-attribute Learning for Pedestrian Attribute Recognition in Surveillance Scenarios" . And you could modify the base net by yourself, like transfer to ResNet or something else, and replace the last layer to multi task layer, and I have used the dataset published by @xh-liu in this paper, so grateful. I hope it helps

**Comment by kclch** [comment]
Can you share your entire program with me? I will be very grateful to you.

**Comment by kclch** [comment]
@JohnHush Can you share your entire program with me? I will be very grateful to you.

**Comment by kclch** [comment]
@JohnHush Hello, I used the CPU during training and found that there is no weight model file "bvlc_reference_caffenet.caffemodel" in "train_PA100K.sh". How can I find "bvlc_reference_caffenet.caffemodel"? Thank you very much.

**Comment by kclch** [comment]
@JohnHush /home/a/图片/2018-09-15 21-57-33屏幕截图.png
Hello, when I reproduced DeepMAR, this problem occurs in "caffe runtest" in caffe. The problem is due to the data storage problem, but I have not started training the model, just compiling caffe, what is the problem? ? thank you very much

**Comment by kclch** [comment]
@JohnHush
![2018-09-15 21-57-33](https://user-images.githubusercontent.com/39482063/45587184-e09fd380-b8f1-11e8-83ca-b545516d891d.png)
Hello, when I reproduced DeepMAR, this problem occurs in "caffe runtest" in caffe. The problem is due to the data storage problem, but I have not started training the model, just compiling caffe, what is the problem? ? thank you very much

**Comment by kclch** [comment]
![multilabel prototxt](https://user-images.githubusercontent.com/39482063/45587378-2d84a980-b8f4-11e8-9b03-4739612a0ef9.png)
And I can't find the "multilabel.prototxt" file

**Comment by kclch** [comment]
@JohnHush Can you provide the test code, thank you
